src/v/datalake/translation/partition_translator.cc

```diff
@@ -201,7 +201,10 @@ ss::future<> partition_translator::translate_until_stopped() {
         // until 'wait_for_data' returns and we re-enter the loop.
         _data_source->update_commit_lag(last_committed_offset);
         _data_source->update_translation_lag(last_translated_offset);
-
+        /**
+         * Update cached lag
+         */
+        _cached_current_lag = co_await current_lag_ms();
```

Comment:
What's the rationale behind caching the lag rather than caching the last translated timestamp? I would've thought that we care more about the timestamp since the actual value for lag will change between here and the next call to `status()`